internal/cephfs/controllerserver.go

```diff
@@ -114,29 +226,51 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+
 	defer func() {
 		if err != nil {
-			errDefer := undoVolReservation(ctx, volOptions, *vID, secret)
-			if errDefer != nil {
-				klog.Warningf(util.Log(ctx, "failed undoing reservation of volume: %s (%s)"),
-					requestName, errDefer)
+			if !errors.As(err, &ecip) {
+				errDefer := undoVolReservation(ctx, volOptions, *vID, secret)
+				if errDefer != nil {
+					klog.Warningf(util.Log(ctx, "failed undoing reservation of volume: %s (%s)"),
+						requestName, errDefer)
+				}
 			}
 		}
 	}()
 
+	// Take lock on parent snapshot or volume
```

Comment:
the lock is taken multiple times on snapshot and clone in CreateVolume isn't it? is it required?

Reply:
not really .. done.. thanks.